### PR TITLE
LPS-196126: Fix link a relationship System Object and a Custom Object Entry

### PR DESCRIPTION
**Pull request by sergiojimcos**
## Purpose
The aim of this PR  is to fix link System Object Entries with Custom Object Entries using the Relationship Endpoints. The problem comes from the Object Definition Service, there is a method called `getSystemObjectDefintions` which will return all System Object Definitions that are modifiable and unmodifiable. 

**What is new?**
- Define new Filter in `service.xml`, given companyId, `modifiable` and `system`.
- Define new method in `ObjectDefintionLocalService`.
- Use the proper method in `RelatedObjectEntryResourceImpl` to obtain all the unmodifiable Object Definitions.

**How to test this?**
1. Set all API Builder FFs
2. Create a custom Object Subject with a custom text field subjectName.
3. Modify the User system object to create a new custom integer field teacherCode.
4. Create a one to many relationship in User that relates User and Subject called teacherSubjects.
5. Create a User with a value in the teacherCode (it could be by using the headless-admin-user REST API) for example:
```curl
curl -X POST 'http://localhost:8080/o/headless-admin-user/v1.0/user-accounts' \ -H 'accept: application/json' \ -H 'Content-Type: application/json' \ -u 'test@liferay.com:test' \ -d '{ "additionalName": "teacher_1", "alternateName": "teacher_1", "emailAddress": "teacher_1@liferay.com", "familyName": "teacher_1", "givenName": "teacher_1", "jobTitle": "teacher_1", "teacherCode": 1106 }'
```
6. Create a subject and relate it to the previously created User

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPS-196126

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -734,6 +734,11 @@ public ObjectDefinition getObjectDefinition(long objectDefinitionId)
 		return objectDefinitionPersistence.findByPrimaryKey(objectDefinitionId);
 	}
 
+	@Override
+	public List<ObjectDefinition> getObjectDefinitions(long companyId) {
```
Comment (4lejandrito) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:738]: Call this `getUnmodifiableSystemObjectDefinitions`. Currently the real behaviour is obscured by the name of the method.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by sergiojimcos** [comment]
ci:foward

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
@sergiojimcos can you write an automated test that reproduces the original issue?

**Comment by sergiojimcos** [comment]
@4lejandrito this will be covered if we activate the FF of the API Builder

**Comment by sergiojimcos** [comment]
ci:forward

**Review by 4lejandrito — APPROVED** [review]

**Comment by magjed4289** [comment]
test failures are not related. OK to manually forward ✅

**Comment by sergiojimcos** [comment]
Sent to Brian here: https://github.com/brianchandotcom/liferay-portal/pull/141397

**Pull request closed without merging**